Objects/codeobject.c

```diff
@@ -1371,7 +1386,20 @@ _PyCode_SetExtra(PyObject *code, Py_ssize_t index, void *extra)
 PyObject *
 _PyCode_GetVarnames(PyCodeObject *co)
 {
-    return get_localsplus_names(co, CO_FAST_LOCAL, co->co_nlocals);
+    if (init_co_cached(co)) {
+        return NULL;
+    }
+    if (co->_co_cached->_co_varnames != NULL) {
```

Comment:
You can factor the body of these functions out. `get_cached_locals()` which takes the code object, the flag, and a pointer to the field to be set.